src/Worker/UpdateAllSuggestions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Worker;
+
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Util\DateTimeFormat;
+
+/**
+ * Update contact suggestions for all aktive users
+ */
+class UpdateAllSuggestions
+{
+	public static function execute()
+	{
+		$users = DBA::select('user', ['uid'], ["`login_date` > ?", DateTimeFormat::utc('now - 7 days')]);
```

Comment:
This currently doesn't include API users, see https://github.com/friendica/friendica/issues/10518

Reply:
Which indicator of activity can we then use additionally? There is the field `last-item` in the contact table. It is set whenever someone created post. We can query this value for the "self" contact as well. So we then could additionally collect all contacts that don't fall in the first category but are at least active in some way.

Reply:
Or we leave this here, but we fix #10518 🥺

Reply:
I wouldn't want to restructure the whole login system at this point. But I guess it would be a viable idea to add a user field with the date of the latest activity. This could be set whenever a person had been active via web or API. Then we can reduce that check to an even shorter time.

Reply:
Do you want to do this in this PR? At the moment, the use of this field would prevent suggestions from being updated for mobile-only users, which is a regression.

Reply:
I could give it a try. So you are okay with the idea of an additional field in the user table?

Reply:
Go for it.